### PR TITLE
init role selection

### PR DESCRIPTION
**Pull request by mb3rnard**
When testing, use "post-install-cmd" for the hook instead of "post-create-project-cmd'
and comment or delete the "removeSymfonyStandardFiles" call.

### CODE REVIEW

File: app/SymfonyStandard/Composer.php [hunk 1]
```diff
@@ -105,12 +103,252 @@ public static function hookCreateProject(CommandEvent $event)
...
+        }
+
+        return $projectInfos;
+    }
+
+    /**
+     * Ask for the vendor.
+     *
+     * @param CommandEvent $event
+     * @return mixed
```
Comment (nervo) [inline app/SymfonyStandard/Composer.php]: We don't know the return type ?

File: app/SymfonyStandard/Composer.php [hunk 2]
```diff
@@ -105,12 +103,252 @@ public static function hookCreateProject(CommandEvent $event)
...
+
+    /**
+     * Handle the dependency version.
+     *
+     * @param $dependency
+     * @param $content
+     * @param $event
+     * @param $vars
+     */
+    private static function handleVersion($dependency, $content, $event, &$vars)
```
Comment (nervo) [inline app/SymfonyStandard/Composer.php]: handleDependencyVersion ?

File: app/SymfonyStandard/Composer.php [hunk 3]
```diff
@@ -105,12 +103,252 @@ public static function hookCreateProject(CommandEvent $event)
...
+        {
+            preg_match('/#'. $dependency .'_version.*\r?\n/', $content, $versionMatches);
+            $vars[$versionMatches[0]] = '';
+        }
+    }
+
+    /**
+     * Handle the dependency version.
+     *
+     * @param $dependency
```
Comment (nervo) [inline app/SymfonyStandard/Composer.php]: Please, set param types here (string, int, ...)

File: app/SymfonyStandard/Composer.php [hunk 4]
```diff
@@ -55,39 +63,29 @@ public static function hookCreateProject(CommandEvent $event)
...
         };
 
-        $vendor = $event->getIO()
-            ->askAndValidate(
-                '<info>Vendor name</info>: ',
-                $validator,
-                1,
-                null
-            );
+        $projectName = self::askAppVendorName($event);
```
Comment (nervo) [inline app/SymfonyStandard/Composer.php]: The method returns an array projectInfos, projecName does not seem relevant

File: app/config/config.yml [hunk 5]
```diff
@@ -28,7 +28,7 @@ framework:
         # http://symfony.com/doc/current/reference/configuration/framework.html#handler-id
         handler_id:  session.handler.native_file
         save_path:   "%kernel.root_dir%/../var/sessions/%kernel.environment%"
-        name:        {{ app.name }}
+        name:        <# app.name #>
```
Comment (nervo) [inline app/config/config.yml]: Could you just set "app" here ?

`name:        app`

File: app/SymfonyStandard/Composer.php [hunk 6]
```diff
@@ -55,39 +63,29 @@ public static function hookCreateProject(CommandEvent $event)
...
-        $appHost = $app . ($vendor ? '.' . $vendor : '');
+        $vendorLabel = $vendor ? ucwords(str_replace('-', ' ', $vendor))  : $appLabel;
+        $appComposerName = str_replace('.', '/', $appName);
 
         $vars = [
-            '{{ vendor }}'       => strtolower($vendor),
-            '{{ app }}'          => strtolower($app),
-            '{{ app_database }}' => strtolower($appDatabase),
-            '{{ app_host }}'     => strtolower($appHost),
+            '<# app.name #>'       => strtolower($appName),
```
Comment (nervo) [inline app/SymfonyStandard/Composer.php]: Could you globaly pass "<#" and "#>" as constants ?

File: app/SymfonyStandard/Composer.php [hunk 7]
```diff
@@ -105,12 +103,252 @@ public static function hookCreateProject(CommandEvent $event)
...
+        $versionReplacement = preg_replace(['/#/', '/\'\d.*\'.*(\r?\n)/'], ['', '\'' . $chosenVersion.'\'$1'] , $versionMatches[0]);
+        $vars[$versionMatches[0]] = $versionReplacement;
+    }
+
+    /**
+     * Get the dependency default version.
+     *
+     * @param $content
+     * @param $dependency
+     * @return mixed
```
Comment (chalasr) [inline app/SymfonyStandard/Composer.php]: PHPDoc sections should be separated by a blank line.

```diff
-     * @param $content
-     * @param $dependency
-     * @return mixed
+     * @param string $content
+     * @param string $dependency
+     *
+     * @return string
+     *
+     * @throws \InvalidArgumentException If no version is found
```

**Pull request merged**